### PR TITLE
Fix/remove department logic

### PR DESCRIPTION
**Pull request by VamshiBatta07**
Removed department and surveyCycleParameter logic.
Updated the logic to conduct a survey.

### CODE REVIEW

File: src/survey/survey.controller.ts [hunk 1]
```diff
@@ -11,37 +11,40 @@ import {
...
-  @Post(":departmentId")
-  @ApiOperation({ summary: "Create Survey-Forms for a department." })
+  @Post(":configId")
+  @ApiOperation({ summary: "Create Survey-Forms for a SurveyConfig." })
   @ApiResponse({ status: HttpStatus.CREATED })
-  async generateSurveyFormsForDepartment(
+  async generateSurveyFormsForSurveyConfig(
     @Res() res,
-    @Param("departmentId") departmentId: number
+    @Param("configId") configId: number
```
Comment (prashantesmagico) [inline src/survey/survey.controller.ts]: Use ParseIntPipe here

File: src/survey/survey.controller.ts [hunk 2]
```diff
@@ -97,11 +100,11 @@ export class SurveyController {
 
   @Get("latest-survey-response/:userId")
   @ApiOperation({ summary: "Fetch latest survey response by userId" })
-  @ApiResponse({ status: HttpStatus.OK })
+  @ApiResponse({ status: HttpStatus.OK, type: ResponseSurveyFormDto })
   async getLatestSurveyResponsesForUserId(
     @Res() res,
     @Param("userId") userId: string
```
Comment (prashantesmagico) [inline src/survey/survey.controller.ts]: use ParseUUIDPipe here

File: src/user-metadata/user-metadata.service.ts [hunk 3]
```diff
@@ -211,13 +206,35 @@ export class UserMetadataService {
...
+    const userValidationArray = [...new Set([...userMapping.assessorIds, userMapping.assesseeId])];
+  
+    const idsNotInUserIdList: string[] = userValidationArray.filter(id => !userIdList.has(id));
+
+    for (const id of idsNotInUserIdList) {
+      try {
+        const user = await prismaClient.userMetadata.findUnique({
+          where: { userId: id },
+        });
+        console.log({user,})
```
Comment (prashantesmagico) [inline src/user-metadata/user-metadata.service.ts]: please remove console

**Review by prashantesmagico — CHANGES_REQUESTED** [review]

**Review by prashantesmagico — APPROVED** [review]

File: src/survey/survey.service.ts [hunk 4]
```diff
@@ -95,63 +101,42 @@ export class SurveyService {
...
-      "SurveyCycleParameters.[0].id",
-      0
-    );
-
-    if (!surveyCycleParameterId || _.isEqual(surveyCycleParameterId, 0)) {
-      throw new BadRequestException(
-        `No any active survey cycle found for department id #${departmentId}`
+    if (_.isUndefined(userMapping) || _.isEmpty(userMapping)) {
+      throw new NotFoundException(
+        `No user mapping found for Survey Ci=onfig with id ${configId}`
```
Comment (prashantesmagico) [inline src/survey/survey.service.ts]: please edit config name

**Review by prashantesmagico — APPROVED** [review]

File: src/survey-config/dto/response-survey-config.dto.ts [hunk 5]
```diff
@@ -1,13 +1,26 @@
...
+}
+
+export class ConfigResponseDTO {
+  readonly data: ResponseSurveyConfigDto;
+  readonly message: string;
+}
+
+export class UserMappingDTO {
+  assesseeId: string;
+  assessorIds: string;
```
Comment (prashantesmagico) [inline src/survey-config/dto/response-survey-config.dto.ts:20]: array of string ID for assessorIds

File: src/response-tracker/response-tracker.service.ts [hunk 6]
```diff
@@ -301,4 +306,35 @@ export class ResponseTrackerService {
...
+    const totalSurveysFilled = await this.getCountByStatus({
+      ...isActiveSurveyCondition,
+      status: ResponseTrackerStatusEnum.COMPLETED,
+    });
+
+    return { totalSurveys, totalSurveysFilled, totalSurveysToBeFilled };
+  }
+
+  async getCountByStatus(whereCondition): Promise<number> {
+    console.log(whereCondition);
```
Comment (prashantesmagico) [inline src/response-tracker/response-tracker.service.ts]: remove console

File: src/survey-config/dto/response-survey-config.dto.ts (hunk 5 above)
Comment (VamshiBatta07) [reply src/survey-config/dto/response-survey-config.dto.ts:20]: This DTO is used for parsing the CSV data where it won't be an array of strings.

**Review by prashantesmagico — APPROVED** [review]

File: src/survey-form/survey-form.service.ts [hunk 7]
```diff
@@ -143,9 +143,33 @@ export class SurveyFormService {
...
     });
   }
+
+  async fetchLatestSurveyScoreByUserId(userId: string): Promise<number|null> {
+    const latestScore =  await this.prisma.surveyForm.findFirst({
+      where: {
+        userId,
+        status: SurveyStatusEnum.CLOSED,
+        SurveyConfig: {
+          isActive: true,
```
Comment (Ajay-Maury) [inline src/survey-form/survey-form.service.ts]: modify query according to form status and config

**Review by Ajay-Maury — APPROVED** [review]

**Pull request merged**